editor/blocks/heading-block/index.js

```diff
@@ -0,0 +1,23 @@
+const { query, html, prop } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'wp/heading', {
+	title: 'Heading',
+	icon: 'heading',
```

Comment:
`heading` isn't a valid dashicon. This might be an example where we'd want to use a custom element as the icon instead of dashicon, because in the mockups it's shown as just an "H" letter ([screenshot](https://wpcoredesign.mystagingwebsite.com/gutenberg/Type%20Switcher.png)). Not sure that's something we'd ever design as an icon (maybe?). In any case, the [current docs outline plans](https://github.com/WordPress/gutenberg/tree/master/blocks#wpblocksregisterblock-slug-string-settings-object-) to support returning an arbitrary element, so this could be: `icon: () => <span>H</span>`

Couple notes for myself:

- Next version of React (16) will support strings as elements, allowing further simplifying to `icon: () => 'H'`
- The docs should probably be updated to allow `icon` to be defined as a function (or even more accurately a component), not necessarily an element